plugin/clang.vim

```diff
@@ -291,6 +299,24 @@ func! s:DiscoverIncludeDirs(clang, options)
   return l:res
 endf
 "}}}
+"{{{ s:DiscoverDefaultIncludeDirs
+" Discover default include directories of clang and gcc (if existed).
+" @options Additional options passed to clang and gcc, e.g. -stdlib=libc++
+" @return List of dirs: ['path1', 'path2', ...]
+func! s:DiscoverDefaultIncludeDirs(options)
+  let l:res = s:DiscoverIncludeDirs(g:clang_exec, a:options)
+  if g:clang_include_sysheaders_from_gcc
+    let l:res += s:DiscoverIncludeDirs(g:clang_gcc_exec, a:options)
+  endif
+  let d = {}
+  for l:dir in l:res
+    let d[l:dir] = ''
+  endfor
+  let l:merged = sort(keys(d))
```

Comment:
Include path order is important, so we can't use `sort` here.
I recommend to select clang *or* gcc to discover headers, not both of them, because we really don't know how to merge them...